src/public/publicAPIs.ts

```diff
@@ -312,4 +312,22 @@ export function navigateToTab(tabInstance: TabInstance, onComplete?: (status: bo
 
   const errorMessage = "Invalid internalTabInstanceId and/or channelId were/was provided";
   GlobalVars.callbacks[messageId] = onComplete ? onComplete : getGenericOnCompleteHandler(errorMessage);
-}
\ No newline at end of file
+}
+
+/**
+ * Sends query to bot in order to retrieve data.
+ * @param botRequest query to send to bot.
+ * @param callback callback to invoke when data is retrieved from bot
+ */
+export function getBotData(botRequest: BotAPIRequest, callback?: (botResponse: any) => void): void { // void for now
```

Comment:
How does this function report errors?

Reply:
Lets use callback?: (status: boolean, data?: string | BotResponse) => void;

1. In case of success
status will be true and data will be response
2. in case of fail
status will be false and data will be response